Chap_API_Sync_Access.tex

```diff
@@ -1,171 +1,422 @@
+However, when providing keys that are documented, they must adhere to the behavior defined by the standard for reserved keys.  
+An implementation cannot provide a reserved key that behaves differently than what is in the standard.  
+Implementations are encouraged to standardize any non-standardized, reserved keys that they provide to avoid conflicting with other implementations or efforts to standardize the same key.
+
+Host environments are permitted to provide non-reserved keys and non-standardized reserved keys.  
+However there are important caveats to providing either of these categories of keys 
+that are explained in \ref{chap:data_sharing:non_rsvd_keys} and \ref{chap:api_rsvd_keys}.
+% Would like to make this statement stronger, but there is text in chapter 6 currently that
+% says it is allowed to use "custom-prefixed keys which may adhere to either the reserved or non-reserved
+% retrieval rules.
+Host environments should avoid providing reserved keys which are not standardized.  
+If the reserved key is later standardized to behave in a way that is different from the behavior 
+provided by the host environment, the host environment will not be compliant with the PMIx standard.
+ 
+Reserved keys are set by the host environment and are made available at client initialization. 
+Each key is associated with a data realm and the retrieval API call will indicate the realm being queried.
+Reserved keys are used to access information about a client's execution context.
+After initialization, the client can access reserved keys using the PMIx_Get API.  
+Common information that a client may access includes, but is not limited to,
+the name of the host it is executing on, 
+the number of peer processes in its namespace,
+the number of peer processes in its application,
+and its job rank.
+
+%%% (Do we want to make any other statements about how and when values can change during 
+%%%  the course of execution.
+%%%  PMIX_EXIT_CODE for example, UNIVERSE_SIZE could change if more resources are added?)
+
+Non-reserved keys are provided by clients.
+A call to PMIx_Put will result in a key-value being associated with the calling client within 
+its process data realm.
+There is no mechanism to retract a key-value that has been made available by a client through \refapi{PMIx_Put}.
+The key can be assigned a new value, but cannot be removed entirely.
+Although key values can originate from different sources, they are always retrieved using the
+\refapi{PMIx_Get} and \refapi{PMIx_Get_nb} API.
+
+The publish/lookup data realm is accessed through a separate set of API's.
+\refapi{PMIx_Put} cannot add or modify key-values within the publish/lookup realm and 
+\refapi{PMIx_Lookup} cannot access key-values outside the publish/lookup realm.
+This data realm is described in detail is chapter [Reference to publish/lookup chapter].
+Although PMIx_Publish and PMIx_Lookup are analogous to PMIx_Put and PMIx_Get in that both pairs 
```

Comment:
Editorial tex updates - note that there are a few more of these in the paragraph that follows (I did not highlight them all)
```suggestion
Although \refapi{PMIx_Publish} and \refapi{PMIx_Lookup} are analogous to \refapi{PMIx_Put} and \refapi{PMIx_Get} in that both pairs 
```